cmd/cloudshell_open/cloudrun.go

```diff
@@ -102,18 +125,57 @@ func promptDeploymentRegion(ctx context.Context, project string) (string, error)
 	return choice, nil
 }
 
-func serviceURL(project, name, region string) (string, error) {
+
+func describe(project, name, region string) (service, error) {
+	var service service
+	var o bytes.Buffer
+	var e bytes.Buffer
+
 	cmd := exec.Command("gcloud", "run", "services", "describe", name,
 		"--project", project,
 		"--platform", "managed",
 		"--region", region,
-		"--format", "value(status.address.url)")
+		"--format=json")
+
+	cmd.Stdout = &o
+	cmd.Stderr = &e
+	if err := cmd.Run(); err != nil {
+		return service, fmt.Errorf("error describing service: %+v. output=\n%s", err, e.String())
+	}
+
+	if err := json.NewDecoder(&o).Decode(&service); err != nil {
+		return service, fmt.Errorf("error decoding gcloud --format=json output: %+v", err)
+	}
+
+	return service, nil
+}
+
+func serviceURL(project, name, region string) (string, error) {
+	service, err := describe(project, name, region)
+
+	if err != nil {
+		return "", err
+	}
+
+	return service.Status.Address.Url, nil
+}
+
+func envVars(project, name, region string) (map[string]struct{}, error) {
+	service, err := describe(project, name, region)
 
-	b, err := cmd.CombinedOutput()
 	if err != nil {
-		return "", fmt.Errorf("deployment to Cloud Run failed: %+v. output:\n%s", err, string(b))
+		return nil, err
+	}
+
+	existing := make(map[string]struct{})
```

Comment:
map[string]bool

Reply:
It appears that `map[string]struct{}` is the connonical way to do a set in Go.  But I'm not certain. Let me know if you'd rather do it via `map[string]bool`

Reply:
yeah it's all the same since you use the keys as a set.